standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/security/ZooKeeperTokenStore.java

```diff
@@ -479,26 +482,50 @@ public void init(Object hmsHandler, HadoopThriftAuthBridge.Server.ServerMode sMo
         conf.get(MetastoreDelegationTokenManager.DELEGATION_TOKEN_STORE_ZK_CONNECT_STR, null);
     if (zkConnectString == null || zkConnectString.trim().isEmpty()) {
       // try alternate config param
-      zkConnectString =
-          conf.get(
-              MetastoreDelegationTokenManager.DELEGATION_TOKEN_STORE_ZK_CONNECT_STR_ALTERNATE,
-              null);
+      zkConnectString = conf.get(MetastoreDelegationTokenManager.DELEGATION_TOKEN_STORE_ZK_CONNECT_STR_ALTERNATE, null);
       if (zkConnectString == null || zkConnectString.trim().isEmpty()) {
-        throw new IllegalArgumentException("Zookeeper connect string has to be specified through "
-            + "either " + MetastoreDelegationTokenManager.DELEGATION_TOKEN_STORE_ZK_CONNECT_STR
-            + " or "
-            + MetastoreDelegationTokenManager.DELEGATION_TOKEN_STORE_ZK_CONNECT_STR_ALTERNATE
-            + WHEN_ZK_DSTORE_MSG);
+        throw new IllegalArgumentException("Zookeeper connect string has to be specified through " + "either "
+            + MetastoreDelegationTokenManager.DELEGATION_TOKEN_STORE_ZK_CONNECT_STR + " or "
+            + MetastoreDelegationTokenManager.DELEGATION_TOKEN_STORE_ZK_CONNECT_STR_ALTERNATE + WHEN_ZK_DSTORE_MSG);
+      }
+      // If we use the alternate zk config (the global one)
+      // we should also use the related properties from the global config
+      zkConnectPort = MetastoreConf.getVar(conf, MetastoreConf.ConfVars.THRIFT_ZOOKEEPER_CLIENT_PORT);
+      connectTimeoutMillis = (int) MetastoreConf
+          .getTimeVar(conf, MetastoreConf.ConfVars.THRIFT_ZOOKEEPER_CONNECTION_TIMEOUT, TimeUnit.MILLISECONDS);
+      sslEnabled = MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.THRIFT_ZOOKEEPER_SSL_ENABLE);
```

Comment:
This is a bit awkward: if (sslEnabled) {} else {sslEnabled=..., if (sslEnabled)...}

Reply:
That else is for the previous if statement.
if (we use the alternate config)
  read the alternate sslEnabled
  if( that property is set)
   read the other properties from the alternate config
else // we use the delegation own config
  read the delagation sslEnabled config
  if (the that property is set)
    read the other properties from the delegation config

Reply:
Thanks!